src/Testcontainers/Builders/DockerDesktopEndpointAuthenticationProvider.cs

```diff
@@ -15,7 +15,7 @@ internal sealed class DockerDesktopEndpointAuthenticationProvider : RootlessUnix
     /// Initializes a new instance of the <see cref="DockerDesktopEndpointAuthenticationProvider" /> class.
     /// </summary>
     public DockerDesktopEndpointAuthenticationProvider()
-      : base(GetSocketPathFromHomeDesktopDir(), GetSocketPathFromHomeRunDir())
+      : base(DockerConfig.Instance.GetCurrentEndpoint()?.AbsolutePath, GetSocketPathFromHomeDesktopDir(), GetSocketPathFromHomeRunDir())
```

Comment:
Why do we use the Docker Desktop provider here? This prevents users from overriding the `DOCKER_HOST` configuration (which is likely irrelevant) but it also prevents them from setting the `TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE` configuration. IIRC, this setup was necessary for Colima. On the test host, the socket path is `$COLIMA_HOME/default/docker.sock`, but in the VM that runs the containers it is a different path. Is it always `/var/run/docker.sock` (I do not remember anymore)?